dialob-components/dialob-form-service-rest/src/main/java/io/dialob/form/service/rest/FormsRestServiceController.java

```diff
@@ -155,6 +164,45 @@ public ResponseEntity<Form> postForm(Form formDocument) {
     return ResponseEntity.created(uri).body(savedForm);
   }
 
+  @Override
+  public ResponseEntity<FormPutResponse> postFormFromCsv(String formCsv) {
+    try {
+      Form formDocument = csvToFormParser.parseCsv(formCsv);
+      Form form = updateMetadata(formDocument);
+      form = ImmutableForm.builder().from(form).id(null).rev(null).build();
+      Form savedForm = formDatabase.save(currentTenant.getId(), form);
+
+      URI uri = ServletUriComponentsBuilder
+        .fromCurrentRequest().path("/{id}")
+        .buildAndExpand(savedForm.getId()).toUri();
+
+      // Success Response
+      return ResponseEntity.created(uri)
+        .body(ImmutableFormPutResponse.builder()
+        .ok(true)
+        .id(savedForm.getId())
+        .rev(savedForm.getRev())
+        .form(savedForm)
+        .build());
+    } catch (CsvParsingException e) {
+      // Bad Request Response
+      return ResponseEntity.badRequest()
+        .body(ImmutableFormPutResponse.builder()
+        .ok(false)
+        .error("CSV_PARSING_ERROR")
+        .reason(e.getMessage())
+        .build());
+    } catch (Exception e) {
+      // Internal Server Error Response
+      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+        .body(ImmutableFormPutResponse.builder()
+        .ok(false)
+        .error("INTERNAL_SERVER_ERROR")
+        .reason("An unexpected error occurred. Error message: " + e.getMessage())
+        .build());
+    }
```

Comment:
Please check if this is necessary, AFAIK any exception thrown from controller method will be handled by global handler as 500